squape/video.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2023, Cyber Alpaca
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+import os
+import shutil
+from contextlib import contextmanager
+from importlib import resources
+from pathlib import Path
+
+import squishinfo
+import test
+
+
+def _failures_results_count() -> int:
+    """
+    This function returns the total number of failures in test results by adding up
+    all the counts for errors, fatals, xpasses and fails.
+
+    Returns:
+        int: The total number of failures in test results.
+    """
+    return (
+        test.resultCount("errors")
+        + test.resultCount("fatals")
+        + test.resultCount("xpasses")
+        + test.resultCount("fails")
+    )
+
+
+def _videos_set() -> set:
+    """
+    This function returns a set of existing video filenames in video directory
+    RESULT_DIR/TEST_SUITE_NAME/TEST_CASE_NAME/attachments.
+
+    Returns:
+        set: The set of filenames with mp4 extension
+    """
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+
+    videos_set = set()
+
+    if not os.path.exists(video_dir):
+        return videos_set
+
+    for file in os.listdir(video_dir):
+        if file.endswith(".mp4"):
+            videos_set.add(file)
+    return videos_set
+
+
+def _remove_videos(videos: set) -> None:
+    """
+    Remove videos from the attachments directory. The original video will be replaced
+    with a placeholder video with size of 10.6 KB.
+
+    Args:
+        videos (set): A set of video names to remove.
+        Does nothing when the set is empty.
+    """
+
+    if not len(videos):
+        return None
+
+    video_dir = os.path.join(
+        squishinfo.resultDir,
+        squishinfo.testSuiteName,
+        squishinfo.testCaseName,
+        "attachments",
+    )
+
+    res_container = Path(resources.files(__package__), Path("resources"))
+    empty_video = Path(res_container, "empty_video_with_message.mp4")
+
+    for video_name in videos:
+        test.log(f"Remove video: {video_name}")
+        orig_video = Path(video_dir, video_name)
+        shutil.copy(empty_video, orig_video)
+
+
+@contextmanager
+def video_capture(message: str = None, remove_on_success: bool = False) -> None:
```

Comment:
Emptry string instead of None as the default for the message would simplify the function

Reply:
Fixed with [95f6c0a](https://github.com/CyberAlpaca/squish-api-python-extension/pull/16/commits/95f6c0a0268b5100a8783db87d1f8d20669c3a76)